### ISSUE TITLE
A Primer in BERTology: What We Know About How BERT Works

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#231]
# Information
### URL
https://aclanthology.org/2020.tacl-1.54/
### Authors
[Anna Rogers](https://aclanthology.org/people/a/anna-rogers/), [Olga Kovaleva](https://aclanthology.org/people/o/olga-kovaleva/), [Anna Rumshisky](https://aclanthology.org/people/a/anna-rumshisky/)
### Published at
TACL 2020

# Reading Motivation
#114 で紹介されていた BERTology (BERT 学) について.
# Summary
* 概要
Transformer-based model が成功を収めているが, なぜうまくいっているかわかっていない. この論文は, 150以上の BERT に関する研究のサーベイ. レビュー対象は, どのように BERT が work するのかの現状での知識, どのような種類の情報を学習しているか, どのようにそれらが表現されているか, 学習時の objective と architecture の一般的な修正, パラメータ数が多い問題, モデル圧縮のアプローチ. その後, 今後の研究の方向性の概要を述べる.

紹介されている研究が多すぎるので, Bold 箇所を列挙し, 雑にメモする.

3  What Knowledge Does BERT Have?

3.1  Syntactic Knowledge
* BERT representations are hierarchical rather than linear
* BERT embeddings encode information about parts of speech, syntactic chunks, and roles.
* syntactic structure is not directly encoded in self-attention weights.
* syntactic information can be recovered from BERT token representations.
* BERT ‘‘naturally’’ learns some syntactic information, although it is not very similar to linguistic annotated resources.
* BERT takes subject-predicate agreement into account when performing the cloze task
* BERT is better able to detect the presence of NPIs (e.g., ‘‘ever’’) and the words that allow their use (e.g., ‘‘whether’’) than scope violations.
* BERT does not ‘‘understand’’ negation and is insensitive to malformed input.
* either BERT’s syntactic knowledge is incomplete, or it does not need to rely on it for solving its tasks.

3.2 Semantic Knowledge
* BERT has some knowledge of semantic roles
* BERT encodes information about entity types, relations, semantic roles, and proto-roles
* BERT struggles with representations of numbers.
* BERT is surprisingly brittle to named entity replacements

3.3 World Knowledge
* BERT struggles with pragmatic inference and role-based event knowledge
* for some relation types, vanilla BERT is competitive with methods relying on knowledge bases
* BERT cannot reason based on its world knowledge.

3.4 Limitations

---

4 Localizing Linguistic Knowledge

4.1 BERT Embeddings
* contextualized.
* distilled contextualized embeddings better encode lexical semantic information
* two random words will on average have a much higher cosine similarity than expected if embeddings were directionally uniform (isotropic).
* BERT’s contextualized embeddings form distinct clusters corresponding to word senses
* the representations of the same word depend on the position of the sentence in which it occurs

4.2 Self-attention Heads
4.2.1 Heads With Linguistic Functions
* some BERT heads seem to specialize in certain types of syntactic relations.
* no single head has the complete syntactic tree information,
* attention weights are weak indicators of subject-verb agreement and reflexive anaphora.

4.2.2 Attention to Special Tokens
* most selfattention heads do not directly encode any non-trivial linguistic information

4.3 BERT Layers
* the lower layers have the most information about linear word order.
* syntactic information is most prominent in the middle layers of BERT.
* conflicting evidence about syntactic chunks.
* The final layers of BERT are the most taskspecific.
* semantics is spread across the entire model

---

5 Training BERT

5.1 Model Architecture Choices
* the number of heads was not as significant as the number of layers.
* changes in the number of heads and layers appear to perform different functions.

5.2 Improvements to the Training Regime
* the benefits of large-batch training
* the model training can be done in a recursive manner

5.3 Pre-training BERT
* alternative training objectives
  *  How to mask
  * What to mask
  * Where to mask
  * Alternatives to masking
  * NSP alternatives
  * Other tasks.
* incorporate explicit linguistic information
* explicitly supplying structured knowledge

5.4 Fine-tuning BERT
* Taking more layers into account
* Two-stage fine-tuning 
* Adversarial token perturbations
* Adversarial regularization 

---

6 How Big Should BERT Be?

6.1 Overparameterization
* all but a few Transformer heads could be pruned without significant losses in performance.
* Positive effect from head disabling

6.2 Compression Techniques
* BERT can be efficiently compressed with minimal accuracy loss
* knowledge distillation framework
* Quantization
* Pruning

6.3 Pruning and Model Analysis

---

7 Directions for Further Research
* Benchmarks that require verbal reasoning.
* Benchmarks for the full range of linguistic competence.
* Developing methods to ‘‘teach’’ reasoning.
* Learning what happens at inference time.

# Memo
* polarity & polarity item (極性 & 極性表現) [Wikipedia](https://ja.wikipedia.org/wiki/%E6%A5%B5%E6%80%A7_(%E8%A8%80%E8%AA%9E%E5%AD%A6))
